src/executor/current_thread/mod.rs

```diff
@@ -584,6 +637,36 @@ impl<'a, P: Park> fmt::Debug for Entered<'a, P> {
     }
 }
 
+// ===== impl Handle =====
+
+/// Handle to spawn a future on the corresponding `CurrentThread` instance
+#[derive(Clone)]
+pub struct Handle {
+    inner: mpsc::UnboundedSender<Box<Future<Item = (), Error = ()> + Send + 'static>>,
+}
+
+// Manual implementation because the Sender does not implement Debug
+impl fmt::Debug for Handle {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        fmt.debug_struct("Handle")
+            .finish()
+    }
+}
+
+impl Handle {
+    /// Spawn a future onto the `CurrentThread` instance corresponding to this handle
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the spawn fails. Failure occurs if the `CurrentThread`
+    /// instance of the `Handle` does not exist anymore.
+    pub fn spawn<F>(&self, future: F)
```

Comment:
For forwards compat, we will probably want to return `Result` here (matching https://docs.rs/tokio/0.1.6/tokio/executor/trait.Executor.html#tymethod.spawn).

Reply:
Done